### PR TITLE
chore(auth): add refresh session sections

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes:
Added the `refresh session` and `refresh session with social providers` sub sections to the [Manage user session](https://docs.amplify.aws/react/build-a-backend/auth/manage-user-session/) page
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx [hunk 1]
```diff
@@ -138,6 +130,49 @@ async function currentSession() {
...
+        },
+      },
+    },
+  },
+})
+```
+
+<Callout>
+
+**Note:** refresh session with social providers will not work when the user has authenticated with a `token` oauth flow, e.g `{ responseType: 'token' }`.
```
Comment (jimblanc) [inline src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx:172]:
```suggestion
**Note:** `forceRefresh` with social providers will not work when the user has authenticated with a `token` OAuth flow, e.g `{ responseType: 'token' }`.
```

**Review by jimblanc — COMMENTED** [review]
Couple nits

File: src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx (hunk 1 above)
Comment (israx) [reply src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx:172]: It is true that `forceRefresh` will not work under those circumstances, but `fetchAuthSession` will also not be able to automatically refresh the session even though the `forceRefresh` option is not passed.

Comment (jimblanc) [reply src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx]: Ah got it! Okay, discard this one, but maybe still change oauth to OAuth

**Review by jimblanc — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx [hunk 2]
```diff
@@ -138,6 +130,49 @@ async function currentSession() {
 </Block>
 </BlockSwitcher>
 
+### Refreshing sessions
+
+The `fetchAuthSession` API will automatically refresh the user's session when tokens have expired and there is a valid `refreshToken`. Additionally, you can also refresh the session by calling the `fetchAuthSession` API with the `forceRefresh` flag enabled.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx:135]:
```suggestion
The `fetchAuthSession` API automatically refreshes the user's session when the authentication tokens have expired and a valid `refreshToken` is present. Additionally, you can also refresh the session explicitly by calling the `fetchAuthSession` API with the `forceRefresh` flag enabled. 
```

File: src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx [hunk 3]
```diff
@@ -138,6 +130,49 @@ async function currentSession() {
...
+    console.log(tokens);
+  } catch (err) {
+    console.log(err);
+  }
+}
+```
+
+### Refreshing sessions with social providers
+
+To refresh the user's session when authenticating with a social provider, you need to use the `code grant` oauth flow in your Amplify configuration.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/auth/manage-user-session/index.mdx:152]:
```suggestion
To refresh the user's session when authenticating with a social provider, you need to use the `code grant` OAuth flow in your Amplify configuration.
```

**Review by renebrandel — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**